packages/fxa-settings/src/pages/Signup/container.spec.tsx

```diff
@@ -0,0 +1,439 @@
+// mock behaviors can be easily applied. Once applied, they can
+// always be overridden as needed.
+function applyMocks() {
+  // TIP - Important! Clear previous mock states
+  jest.resetAllMocks();
+  jest.restoreAllMocks();
+
+  // Run default mocks
+  mockIntegration();
+  mockServiceName();
+  mockSignupModule();
+  mockApolloClientModule();
+  mockModelsModule();
+  mockUseValidateModule();
+  mockFirefoxModule();
+  mockCryptoModule();
+  mockReachRouterModule();
+  mockReactUtilsModule();
+  mockLoadingSpinnerModule();
+}
+
+// TIP - Since render looks more or less the same each time, we can tease this out
+// as a helper function.
+async function render(text?: string) {
+  // Even though we aren't testing presentation, there still might be presentation
+  // state/context driven by react or l10n that's required. Therefore we will invoke the
+  // container component as follows.
+  renderWithLocalizationProvider(
+    <SingupContainer
+      {...{
+        integration,
+        serviceName,
+      }}
+    />
+  );
+
+  // TIP - Wait for the expected mocked test to show up.
+  await screen.findByText(text || 'sign up mock');
+
+  // TIP/HACK -To work around the fact that the container uses, requestAnimationFrame...
+  // Turns out that can create some pretty erratic test behavior...
+  await new Promise((r) => setTimeout(r, 100));
+}
+
+// TIP - It's easier to target test suites that don't have spaces in their names.
+describe('sign-up-container', () => {
+  // TIP - Always run applyMocks, this will fire before every test.
+  beforeEach(() => {
+    applyMocks();
+  });
+
+  // TIP - Using nested describe is often helpful. It makes the test read easier
+  // and creates a natural partitioning of testing cases that are typically
+  // easier to understand than a bunch of individual tests. They are also a
+  // easier to focus and skip.
+  describe('default-state', () => {
+    it('renders', async () => {
+      await render();
+      expect(screen.queryByText('loading spinner mock')).toBeNull();
+      expect(SignupModule.Signup).toBeCalled();
+      expect(currentSignupProps?.isSync).toBeTruthy();
+    });
+  });
+
+  describe('loading-states', () => {
+    beforeEach(() => {
+      // TIP - In this case, we want to override the previous behavior. We can do this
+      // easily in a before each or even within a test block.
+      (useAuthClient as jest.Mock).mockImplementation(() => {
+        let client = new AuthClient('localhost:9000');
+        client.accountStatusByEmail = jest
+          .fn()
+          .mockReturnValue(new Promise(() => {}));
+        return client;
+      });
+    });
+
+    it('shows loading util account status query resolves', async () => {
+      await render('loading spinner mock');
+      expect(screen.queryByText('sign up mock')).toBeNull();
+    });
+  });
+
+  describe('error-states', () => {
+    beforeEach(() => {
+      jest.spyOn(ApolloModule, 'useMutation').mockReturnValue([
```

Comment:
Melikes this over `<MockedProvider mocks={[mySuccessStateMockRequiringLotsOfBoilerPlate, myFailingMockRequiringLotsOfBoilerPlate]}>` 🙌